http-api/src/lib.rs

```diff
@@ -475,7 +520,27 @@ async fn project_root_handler(ctx: Context) -> Result<Json, Rejection> {
     Ok(warp::reply::json(&projects))
 }
 
-async fn project_handler(ctx: Context, urn: Urn) -> Result<Json, Rejection> {
+async fn project_urn_handler(ctx: Context, urn: Urn) -> Result<Json, Rejection> {
+    let info = project_info(urn, ctx.paths)?;
+
+    Ok(warp::reply::json(&info))
+}
+
+async fn project_alias_handler(ctx: Context, name: String) -> Result<Json, Rejection> {
+    let namestrings = name.to_string();
+    tracing::debug!("looking for project alias {:?}", namestrings);
+
+    let mut aliases = ctx.aliases.write().await;
+    if !aliases.contains_key(&namestrings) {
+        // If the alias does not exist, rebuild the cache.
+        ctx.populate_aliases(&mut aliases).await?;
+    }
+    let urn = aliases
+        .get(&namestrings)
+        .cloned()
+        .ok_or_else(warp::reject::not_found)?;
+    tracing::debug!("project alias resolved to {}", urn);
+
     let info = project_info(urn, ctx.paths)?;
 
     Ok(warp::reply::json(&info))
```

Comment:
It would be clearer here to call `project_urn_handler`, that way the alias handler is only handling the alias resolution.

Reply:
I think we can also get rid of the debug statements in here, as it's pretty easy to know what's going on by just calling the route.